subtests/docker_cli/liverestore/liverestore.py

```diff
@@ -0,0 +1,200 @@
+            docker_cmdline = docker_daemon.cmdline()
+            self.failif_not_in("--live-restore", docker_cmdline,
+                               "docker daemon command-line options",
+                               DockerTestNAError)
+
+        # rhbz1424709#c40 : live-restore fails unless KillMode=process
+        killmode = docker_daemon.systemd_show('KillMode')
+        if killmode != 'process':
+            self.logwarning("systemd KillMode is '%s' (expected 'process')."
+                            " This test will probably fail." % killmode)
+
+        self.stuff['dc'] = DockerContainers(self)
+
+    def run_once(self):
+        super(liverestore, self).run_once()
+
+        self._start_container()
+
+        # Verify that the container is running and that it's a child
+        # of the main docker daemon
+        self._verify_that_date_changes()
+        self.failif(not self._container_is_child_of_docker_daemon(),
+                    "Newly-created container is not a descendant of dockerd")
+
+        # Restart docker daemon. It should be nearly instantaneous; if it
+        # takes ~90 seconds it could be another symptom of rhbz1424709.
+        # The warning message might help someone diagnose a later failure.
+        self.stuff['dockerd_pid_orig'] = docker_daemon.pid()
+        self.stuff['container_pid_orig'] = self._container_pid()
+        t0 = time.time()
+        docker_daemon.restart()
+        t1 = time.time()
+        if t1 - t0 > 30:
+            self.logwarning("docker restart took %d seconds", t1 - t0)
+
+    def _start_container(self):
+        """
+        Start a container. We only need it briefly, until we (the test,
+        running in host space) can cd into its root filesystem. Container
+        will spin idly until a semaphore file is removed.
+        """
+        c_name = self.stuff['dc'].get_unique_name()
+
+        fin = DockerImage.full_name_from_defaults(self.config)
+        subargs = ['--detach', '--name=' + c_name, fin,
+                   'bash -c "echo READY;'
+                   ' while :; do date +%s; sleep 1;done"']
+        dkrcmd = AsyncDockerCmd(self, 'run', subargs)
+        dkrcmd.execute()
+        dkrcmd.wait_for_ready(c_name)
+        self.stuff['container_name'] = c_name
+
+    def _verify_that_date_changes(self):
+        """
+        Verify that container process is running, by running 'docker log'
+        on it sequentially and making sure that (1) we get new output,
+        and (2) the new output corresponds to current time.
+        """
+        log_0 = self._container_log()
+
+        # Wait up to 5 seconds for output to change
+        end_at = time.time() + 5
+        while time.time() <= end_at:
+            log_1 = self._container_log()
+            if log_1 != log_0:
+                break
+            time.sleep(1)
+
+        self.failif(len(log_0) == len(log_1),
+                    "Container log line count did not grow")
```

Comment:
I think in all cases, it would be helpful to have the output for later reference.  Could we do something like this:

```python
        ....
        self.logdebug(self._container_log())
        self.failif(len(log_0) == len(log_1),
        ...
```